library/Zend/InputFilter/Input.php

```diff
@@ -326,6 +326,10 @@ public function isValid($context = null)
             return true;
         }
 
+        if ($empty && !$this->isRequired()) {
```

Comment:
I agree with @Martin-P here; the `isRequired()` method is defined to return a boolean, so we can safely use the `!` operator here without doing a strict comparison.

Strict comparisons are primarily of use when type may be in question; it's not here.